File: src/android/AndroidIntro/AndroidIntro/src/main/java/com/androidIntro/kittenviewer/ListViewDemoActivity.java
package com.androidIntro.kittenviewer;

import android.app.Activity;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ListView;

import java.util.Random;

/**
 * Created by somya on 8/18/13.
 */
public class ListViewDemoActivity extends Activity
{
	@Override
	protected void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);

		setContentView(R.layout.list_view_demo);

		ListView demoListView = (ListView) findViewById(R.id.demo_list_view);

		demoListView.setAdapter(new DemoListAdapter());

	}

	private class DemoListAdapter extends BaseAdapter
	{

		private final LayoutInflater m_layoutInflater;

		private DemoListAdapter()
		{
			m_layoutInflater = (LayoutInflater) getSystemService(LAYOUT_INFLATER_SERVICE);
		}

		@Override public int getCount()
		{
			return 100;
		}

		@Override public Object getItem(final int position)
		{
			return null;
		}

		@Override public long getItemId(final int position)
		{
			return position;
		}

		@Override public View getView(final int position, final View convertView, final ViewGroup parent)
		{
			View listItem = convertView;
			if (null == listItem)
			{
				Log.d("DemoListAdapter", "Creating view for position " + position);

				listItem = m_layoutInflater.inflate(R.layout.demo_list_item_view, parent, false);
			}

			View random_color_view = listItem.findViewById(R.id.random_color_view);

			Random rnd = new Random();
			int color = Color.argb(255, rnd.nextInt(256), rnd.nextInt(256), rnd.nextInt(256));
			random_color_view.setBackgroundColor(color);

			return listItem;
		}
	}
}

File: src/MonoAndroid/AndroidIntro/ListViewDemoActivity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.Util;
using Android;
using Android.Graphics;

namespace AndroidIntro
{
	[Activity (Label = "ListViewDemoActivity")]			
	public class ListViewDemoActivity : Activity
	{
		protected override void OnCreate (Bundle bundle)
		{
			base.OnCreate (bundle);

			SetContentView (Resource.Layout.ListViewDemo);

			FindViewById<ListView> (Resource.Id.demo_list_view).Adapter = new DemoListAdapter(this);



		}
	}

	class DemoListAdapter : BaseAdapter
	{

		private LayoutInflater m_layoutInflater;

		public DemoListAdapter (Context context)
		{
			m_layoutInflater = (LayoutInflater) context.GetSystemService(Context.LayoutInflaterService);
		}

		public override int Count {
			get {
				return 100;
			}
		}

		public override long GetItemId (int position)
		{
			return position;
		}

		public override Java.Lang.Object GetItem (int position)
		{
			return null;
		}

		public override View GetView (int position, View convertView, ViewGroup parent)
		{
			View listItem = convertView;
			if (null == listItem) {
				Log.Debug ("DemoListAdapter", "Creating view for position " + position);

				listItem = m_layoutInflater.Inflate (Resource.Layout.ListViewItemDemo, parent, false);
			}

			View random_color_view = listItem.FindViewById (Resource.Id.random_color_view);
			
			Random rnd = new Random ();
			int color = Color.Argb (255, rnd.Next (256), rnd.Next (256), rnd.Next (256));
			random_color_view.SetBackgroundColor(new Color(color));

			return listItem;
		}
	}
}

